test/Microsoft.AspNetCore.Routing.Tests/EndpointHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    internal static class EndpointHelper
+    {
+        public static MatcherEndpoint CreateMatcherEndpoint(
+            string template,
+            object defaults = null,
+            object requiredValues = null,
+            object constraints = null,
+            int order = 0,
+            string routeName = null,
+            EndpointMetadataCollection metadataCollection = null)
```

Comment:
Suggestions here to make this a little more general purpose.

Make the last parameter of this be `params object[] metadata`.
Also, suggestion, don't include route name here. It's pretty specific to one kind of test, and we can just inline `new RouteNameMetadata` there.